typescript/infra/src/utils/rpcUrls.ts

```diff
@@ -0,0 +1,139 @@
+import { confirm } from '@inquirer/prompts';
+import { ethers } from 'ethers';
+
+import {
+  getSecretRpcEndpoints,
+  getSecretRpcEndpointsLatestVersionName,
+  secretRpcEndpointsExist,
+  setSecretRpcEndpoints,
+} from '../agents/index.js';
+
+import { disableGCPSecretVersion } from './gcloud.js';
+import { isEthereumProtocolChain } from './utils.js';
+
+export async function testProviders(rpcUrlsArray: string[]): Promise<boolean> {
+  let providersSucceeded = true;
+  for (const url of rpcUrlsArray) {
+    const provider = new ethers.providers.StaticJsonRpcProvider(url);
+    try {
+      const blockNumber = await provider.getBlockNumber();
```

Comment:
I think it happened to us recently where a provider would just hang - now that we'll be doing many chains in a single command, may be nice to timeout after a couple secs https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/a49f52cc9c6ab7efa092d52adfa61b724a6f9cb2/typescript/utils/src/async.ts#L17. But don't feel obligated to add now if you'd rather just see if it becomes an issue later

Reply:
na that's a good call, haven't run into it myself but worth adding